### PR TITLE
Feature/25629 colour contrast settings

### PR DESCRIPTION
**Pull request by danwilliams**

### CODE REVIEW

File: pupil-spa/src/app/familiarisation-colour/familiarisation-colour.component.ts [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
+    this.selectedContrast = selectedContrast;
+  }
 
   onClick() {
-    this.router.navigate(['sign-in-success']);
+    const accessArrangements = new AccessArrangements;
+    accessArrangements.contrast = this.selectedContrast || 'bow';
+    this.storageService.setItem(accessArrangementsDataKey, accessArrangements);
+
+    this.router.navigate(['access-settings']);
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/familiarisation-colour/familiarisation-colour.component.ts]: @danwilliams I would expect the navigation to differ for the two following scenarios:
- User is choosing colour contrast pref after login
- User is editing colour contrast pref from access settings page

File: pupil-spa/src/app/familiarisation-colour/familiarisation-colour.component.ts [hunk 2]
```diff
@@ -1,15 +1,42 @@
...
+  selectionChange(selectedContrast) {
+    this.selectedContrast = selectedContrast;
+  }
 
   onClick() {
-    this.router.navigate(['sign-in-success']);
+    const accessArrangements = new AccessArrangements;
+    accessArrangements.contrast = this.selectedContrast || 'bow';
+    this.storageService.setItem(accessArrangementsDataKey, accessArrangements);
+    if (this.routeService.getPreviousUrl() === '/access-settings') {
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply pupil-spa/src/app/familiarisation-colour/familiarisation-colour.component.ts]: @GeorgeChatzigiannis i'm not sure about 'over engineered', but a dedicated access arrangements service is definitely preferable over just adding items into a generic key value storage container.  Maybe the page load hook you suggest is a good idea, but is it really an issue if the settings don't exist?

**Comment by GuyHarwood** [comment]
@activemq is it just the travis issue that is holding this PR up?

**Comment by activemq** [comment]
@GuyHarwood yep, now that it is passing, i will merge.

**Pull request merged**